File: ruby/find_n_unique_integers_sum_up_to_zero.rb
# https://leetcode.com/problems/find-n-unique-integers-sum-up-to-zero/

def sum_zero(n)
  output = []

  if n.odd?
    output << 0

    n -= 1
  end

  n /= 2

  n.times do |i|
    i += 1
    output << i
  end

  n.times do |i|
    i += 1
    output << i * -1
  end

  output
end

File: go/find_n_unique_integers_sum_up_to_zero.go
// https://leetcode.com/problems/find-n-unique-integers-sum-up-to-zero/

package main

import (
	"fmt"
)

func sumZero(n int) []int {
	result := []int{}

	if n%2 != 0 {
		result = append(result, 0)
		n -= 1
	}

	n = n / 2

	for i := 1; i <= n; i++ {
		result = append(result, i) // positive integer
		result = append(result, i*-1) // negative integer
	}

	return result
}

func main() {
	fmt.Println(sumZero(5))
}
